### PR TITLE
PERF: Improve performance for array equal fast

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

gives us a 50% boost if everything is equal and reduces it to nanoseconds if the first non-equal element is within the first couple elements compared to 500 microseconds

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -650,22 +650,20 @@ ctypedef fused int6432_t:
 
 @cython.wraparound(False)
 @cython.boundscheck(False)
-def array_equal_fast(
-    ndarray[int6432_t, ndim=1] left, ndarray[int6432_t, ndim=1] right,
-) -> bool:
+def indexer_equal_fast(ndarray[int6432_t, ndim=1] left, int n) -> bool:
```
Comment (mroeschke) [inline pandas/_libs/lib.pyx:653]: While renaming this function, maybe `is_range_indexer` might be more appropriate? Or at least clarifying that this only compares if the indexer is equal to `range(n)`

Comment (phofl) [reply pandas/_libs/lib.pyx:653]: Yeah thought about this myself and agree.

Comment (phofl) [reply pandas/_libs/lib.pyx]: Changed

**Review by jorisvandenbossche — APPROVED** [review]
Yes, looks good! (that's what I had in mind as well)

**Pull request merged**